lib/vnet/daemon/client_darwin.go

```diff
@@ -0,0 +1,163 @@
+	defer func() {
+		C.free(unsafe.Pointer(result.error_description))
+	}()
+
+	C.RegisterDaemon(&result)
+
+	if !result.ok {
+		status := serviceStatus(result.service_status)
+		// Docs for [registerAndReturnError][1] don't seem to cover a situation in which the user adds
+		// the launch daemon for the first time. In that case, that method returns "Operation not
+		// permitted" error (Code=1, Domain=SMAppServiceErrorDomain). That error doesn't seem to exist
+		// in [Service Management Errors][2]. However, all this means is that the launch daemon was
+		// added to the login items and the user must now enable the corresponding login item.
+		//
+		// We can confirm this by looking at the returned service status.
+		//
+		// [1]: https://developer.apple.com/documentation/servicemanagement/smappservice/register()?language=objc
+		// [2]: https://developer.apple.com/documentation/servicemanagement/service-management-errors?language=objc
+		if status == serviceStatusRequiresApproval {
+			log.DebugContext(ctx, "Daemon successfully added, but it requires approval",
+				"ignored_error", C.GoString(result.error_description))
+			return status, nil
+		}
+
+		log.DebugContext(ctx, "Registering the daemon has failed", "service_status", status)
+		return 0, trace.Errorf("registering daemon failed, %s", C.GoString(result.error_description))
+	}
+
+	return serviceStatus(result.service_status), nil
+}
+
+const waitingForEnablementTimeout = time.Minute
+
+var waitingForEnablementTimeoutExceeded = errors.New("the login item was not enabled within the timeout")
+
+// waitForEnablement periodically checks if the status of the daemon has changed to
+// [serviceStatusEnabled]. This happens when the user approves the login item in system settings.
+func waitForEnablement(ctx context.Context) error {
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	ctx, cancel := context.WithTimeoutCause(ctx, waitingForEnablementTimeout, waitingForEnablementTimeoutExceeded)
+	defer cancel()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return context.Cause(ctx)
+		case <-ticker.C:
+			switch status := daemonStatus(); status {
+			case serviceStatusEnabled:
+				return nil
+			case serviceStatusRequiresApproval:
+				// Continue waiting for the user to approve the login item.
+			case serviceStatusNotRegistered, serviceStatusNotFound:
+				// Something happened to the service since we started waiting, abort.
+				return trace.Errorf("encountered unexpected service status %q", status)
+			default:
+				return trace.Errorf("encountered unknown service status %q", status)
+			}
+		}
+	}
+}
+
+func daemonStatus() serviceStatus {
+	return serviceStatus(C.DaemonStatus())
+}
+
+type serviceStatus int
+
+// https://developer.apple.com/documentation/servicemanagement/smappservice/status-swift.enum?language=objc
+const (
+	serviceStatusNotRegistered serviceStatus = iota
+	serviceStatusEnabled
+	serviceStatusRequiresApproval
+	serviceStatusNotFound
+)
```

Comment:
I'd rather give these explicit values than use iota if we are matching an external definition for them

Reply:
This was me  😅 

Rafal, feel free to revert to your original version.